cwl/src/main/scala/wdl4s/cwl/CwlWomExpression.scala

```diff
@@ -2,67 +2,71 @@ package wdl4s.cwl
 
 import cats.syntax.validated._
 import lenthall.validation.ErrorOr.ErrorOr
-import wdl4s.wdl.types.WdlType
-import wdl4s.wdl.values.{WdlFile, WdlGlobFile, WdlInteger, WdlSingleFile, WdlValue}
+import wdl4s.cwl.CwlWomExpression.EnhancedParameterContextInputs
+import wdl4s.wdl.types.{WdlMapType, WdlNothingType, WdlStringType, WdlType}
+import wdl4s.wdl.values.{WdlFile, WdlMap, WdlString, WdlValue}
 import wdl4s.wom.expression.{IoFunctionSet, WomExpression}
 
-import scala.concurrent.Await
-import scala.concurrent.duration.Duration
+sealed trait CwlWomExpression extends WomExpression {
+  override def evaluateType(inputTypes: Map[String, WdlType]): ErrorOr[WdlType] = ???
```

Comment:
if this is a trait just leave abstract rather than put `???` everywhere

Reply:
None of the implementers currently provide implementations.